sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -189,8 +189,17 @@ case class AdaptiveSparkPlanExec(
           stagesToReplace = result.newStages ++ stagesToReplace
           executionId.foreach(onUpdatePlan(_, result.newStages.map(_.plan)))
 
+          // SPARK-33933: we should submit tasks of broadcast stages first, to avoid waiting
+          // for tasks to be scheduled and leading to broadcast timeout.
+          val reorderedNewStages = result.newStages
```

Comment:
Will this promise be flaky? Does the order of calling `materialize` guarantee that the order of scheduled task?

Reply:
As for flaky, I admit a little bit. 
I believe the order of calling `materialize` can guarantee that the order of task to be scheduled in normal circumstances, but, to be honest, this is guarantee is not strict since the submit of broadcast stage and shuffle map stage are in different thread. But, at least we reach the same level as non AQE.

Reply:
I am also considering a hard guarantee, but it seems the changes are too large.

Reply:
Yea, in non-AQE we always run broadcast jobs first.